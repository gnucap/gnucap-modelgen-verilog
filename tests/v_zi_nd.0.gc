load ../vams/vsine.so
verilog

va_zi_nd #(.d0(1.), .d1(0.), .n0(1.), .n1(1), .T(1)) b(0,out1,in,0);
va_zi_nd #(.d0(1.), .d1(0.), .n0(0.), .n1(1), .T(1)) b(0,out2,in,0);
vsource #(.dc(1.)) v1(in,0);

list

print tran v(out*) control(0) iter(0)
tran 5 basic


vsine #(.mag(1)) vac(in,0);
print ac vm(in) vm(out*) vp(out*)
delete v1

// BUG?: need dc.
dc
ac .001 1 * 1.4

status notime
