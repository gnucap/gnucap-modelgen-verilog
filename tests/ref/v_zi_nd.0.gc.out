va_zi #(.n0(1.),.n1(1),.d0(1.),.d1(0.)) b (.out0(0),.out1(out1),.in0(in),.in1(0));
va_zi #(.n0(0.),.n1(1),.d0(1.),.d1(0.)) b (.out0(0),.out1(out2),.in0(in),.in1(0));
vsource #(.dc(1.)) v1 (.p(in),.n(0));
#Time       v(out1)    v(out2)    control(0)
 0.         0.         0.         1.        
 0.05       0.5        0.         9.        
 0.1        1.         0.         7.        
 1.         1.         0.         2.        
 1.1        2.         1.         7.        
 2.         2.         1.         2.        
 2.1        2.         1.         7.        
 3.         2.         1.         2.        
 3.1        2.         1.         7.        
 4.         2.         1.         2.        
 4.1        2.         1.         7.        
 5.         2.         1.         1.        
#          
 0.        
#Freq       vm(in)     vm(out1)   vm(out2)   vp(out1)   vp(out2)  
 0.001      1.         2.         1.        -0.198     -0.378     
 0.0014     1.         2.         1.        -0.2772    -0.5292    
 0.00196    1.         2.         1.        -0.38808   -0.74088   
 0.002744   1.         1.9999     1.        -0.54331   -1.0372    
 0.0038416  1.         1.9999     1.        -0.76064   -1.4521    
 0.0053782  1.         1.9997     1.        -1.0649    -2.033     
 0.0075295  1.         1.9994     1.        -1.4908    -2.8462    
 0.010541   1.         1.9989     1.        -2.0872    -3.9846    
 0.014758   1.         1.9979     1.        -2.9221    -5.5785    
 0.020661   1.         1.9958     1.        -4.0909    -7.8099    
 0.028925   1.         1.9917     1.        -5.7272    -10.934    
 0.040496   1.         1.9838     1.        -8.0181    -15.307    
 0.056694   1.         1.9684     1.        -11.225    -21.43     
 0.079371   1.         1.9381     1.        -15.716    -30.002    
 0.11112    1.         1.8794     1.        -22.002    -42.003    
 0.15557    1.         1.7659     1.        -30.802    -58.805    
 0.2178     1.         1.5498     1.        -43.123    -82.327    
 0.30491    1.         1.1505     1.        -60.373    -115.26    
 0.42688    1.         0.4554     1.        -84.522    -161.36    
 0.59763    1.         0.60386    1.         61.669     134.1     
 0.83668    1.         1.7425     1.         14.337     43.734    
 1.1714     1.         1.7171     1.        -51.928    -82.772    
Gnucap   System status
iterations: op=0, dc=3, tran=24, fourier=0, total=61
transient timesteps: accepted=12, rejected=0, total=12
nodes: user=4, subckt=0, model=0, total=4
dctran density=50.0%, ac density=62.5%
